src/Raven.Client/Documents/Operations/ETL/AI/AiConnectionString.cs

```diff
@@ -0,0 +1,167 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Raven.Client.Documents.Operations.ConnectionStrings;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.ETL.AI;
+
+public sealed class AiConnectionString : ConnectionString
+{
+    private string _identifier;
+
+    /// <summary>
+    /// A unique identifier used in document paths.
+    /// Only English letters, numbers and hyphens are allowed.
+    /// If not specified, will be auto-generated from the connection name.
+    /// </summary>
+    public string Identifier
+    {
+        get => _identifier ?? GenerateIdentifier(Name);
```

Comment:
I think this needs to be handled server side on Put and validation should be part of the ValidateImpl

Reply:
If we want to guarantee Identifier uniqueness, we should handle both its generation (when the user hasn't provided their own) and validation in the cluster.